### PR TITLE
fix(compiler): != deep inequality

### LINKED ISSUE #3831: != and == have different equality behavior
### I tried this:

Comparing structs using `!=`

### This happened:

Didn't work as expected

### I expected this:

A deep comparison of the fields

### Is there a workaround?

_No response_

### Component

Compiler

### Wing Version

0.25.4

### Node.js Version

18.14.1

### Platform(s)

MacOS

### Anything else?

https://github.com/winglang/wing/blob/71b9b5db4df10c3addeda49032b6925b0ae7bc36/libs/wingc/src/jsify.rs#L601-L604

### Community Notes

* Please vote by adding a 👍 reaction to the issue to help us prioritize.
* If you are interested to work on this issue, please leave a comment.

### PR DESCRIPTION
**Pull request by marciocadev**
The comparison below didn't show an error because it treated them as different objects. Now, the values will be compared, resulting in an error

```ts (wing)
let cat1 = Cat { name: "Mittens", age: 3 };
let cat2 = Cat { name: "Mittens", age: 3 };
assert(cat1 != cat2);
```

Closes #3831 

## Checklist

- [x] Title matches [Winglang's style guide](https://www.winglang.io/contributing/start-here/pull_requests#how-are-pull-request-titles-formatted)
- [x] Description explains motivation and solution
- [x] Tests added (always)
- [ ] Docs updated (only required for features)
- [ ] Added `pr/e2e-full` label if this feature requires end-to-end testing

*By submitting this pull request, I confirm that my contribution is made under the terms of the [Wing Cloud Contribution License](https://github.com/winglang/wing/blob/main/CONTRIBUTION_LICENSE.md)*.

### CODE REVIEW

File: examples/tests/valid/deep_equality.w [hunk 1]
```diff
@@ -36,6 +36,11 @@ test "Json with the same value" {
 
 test "Json with different values" {
   assert(jsonA != jsonC);
+  try {
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:39]:
```suggestion
  assert(!(jsonA != jsonB));
```

Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:39]:
```suggestion
  try {
```

File: examples/tests/valid/deep_equality.w [hunk 2]
```diff
@@ -36,6 +36,11 @@ test "Json with the same value" {
...
+    assert(jsonA != jsonB);
+  } catch {
+    assert(true);
+  }
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:43]:
```suggestion
  assert(!(jsonA != jsonB));

```

Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:43]:
```suggestion
  assert(!(jsonA != jsonB));
```

File: examples/tests/valid/deep_equality.w [hunk 3]
```diff
@@ -53,6 +58,11 @@ test "Set types with the same value" {
 
 test "Set types with different values" {
   assert(setA != setC);
+  try {
+    assert(setA != setB.copy());
+  } catch {
+    assert(true);
+  }
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:65]:
```suggestion
  assert(!(setA != setB.copy()));
```

File: examples/tests/valid/deep_equality.w [hunk 4]
```diff
@@ -106,4 +126,9 @@ test "Struct with the same value" {
 
 test "Struct with different values" {
   assert(cat1 != cat3);
+  try {
+    assert(cat1 != cat2);
+  } catch {
+    assert(true);
+  }
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:133]:
```suggestion
  assert(!(cat1 != cat2));
```

File: examples/tests/valid/deep_equality.w [hunk 5]
```diff
@@ -85,6 +100,11 @@ test "Array with the same value" {
 
 test "Array with different values" {
   assert(arrayA != arrayC);
+  try {
+    assert(arrayA != arrayB.copy());
+  } catch {
+    assert(true);
+  }
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:107]:
```suggestion
  assert(!(arrayA != arrayB.copy()));
```

File: examples/tests/valid/deep_equality.w [hunk 6]
```diff
@@ -69,6 +79,11 @@ test "Map with the same value" {
 
 test "Map with different values" {
   assert(mapA != mapC);
+  try {
+    assert(mapA != mapB.copy());
+  } catch {
+    assert(true);
+  }
```
Comment (marciocadev) [inline examples/tests/valid/deep_equality.w:86]:
```suggestion
  assert(!(mapA != mapB.copy()));
```

**Review by MarkMcCulloh — APPROVED** [review]
🎉

**Pull request merged**